feat: update easy-template-x dependency and remove json flatten function

Sjoertjuh opened the pull request:
This PR updates the easy-template-x dependency and removes the `flattenJSON` function. 
The reason for removing the function is that some easy-template-x's plugins, like the link plugin, require non-flattened json objects. Also, I couldn't really find a reason why it was used anyways. If it is really needed please let me know, so we can think of a better way to implement the plugin features.

bramkn commented:
Hi Sjoert,

I do not know why the FlattenJSON was done, as it has been a while since I created it.
I do not know if this is going to be a breaking change. So unless you put it in as a new version or show me it is not a breaking change of the node I am unable to merge this PR.

Sjoertjuh commented:
Hey, thanks for the feedback. I updated the version in `package.json` and added version history in the readme. Let me know if I need to do anything else for this to be implemented!

bramkn commented:
HI Sjoert,

Sorry, I might not have been clear.
n8n allows for multiple versions to be included in 1 node package, this makes sure that existing flows do not break when a node is updated. So it is not simply changing the package version, it is transforming the node to have multiple versions and then adding the version you created.
The HTTP node and Merge nodes should have this for sure if you want to have a look at examples.

Sjoertjuh commented:
Ah I think I understand now, I'm still pretty new to this all. I now used the ["Light versioning"](https://docs.n8n.io/integrations/creating-nodes/build/reference/node-versioning/#light-versioning), as explained by the docs, since I believe that will suffice. It's not like the Merge node, since that uses ["Full versioning"](https://docs.n8n.io/integrations/creating-nodes/build/reference/node-versioning/#full-versioning).

bramkn commented:
No idea if that is going to be good enough. never tried it.
